packages/openai/src/openai-chat-settings.ts

```diff
@@ -35,6 +35,21 @@ export interface OpenAIChatSettings {
    */
   logitBias?: Record<number, number>;
 
+  /**
+   * Whether to include the log probabilities on the logprobs output field.
+   * 
+   * Including logprobs will increase the response size and can slow down
+   * response times. However, it can be useful for developers to better
+   * understand how the model is behaving.
+   */
+  logprobs?: boolean;
```

Comment:
Can be inferred by `!(topLogprobs === 0 || topLogprobs == null)`. We could e.g. rename `topLogprobs` to logprobs and keep the semantics of `topLogprobs`

Reply:
To be clear, I believe Anthropic, Google and Mistral support logprobs.

Reply:
I'm in general in favor of adding it to the default settings. However, I'm concerned that it is very OpenAI specific atm, so it might be better to wait (so we don't end up with some idiosyncratic). I haven't seen it in the other APIs - can you point me to examples?

Reply:
Makes sense. I think I'm actually mistaken, it seems those other providers don't currently support logprobs.
**Anthropic**: Not supported
**Google**: Supported on PaLM 2 (both text and chat) but not Gemini
**Mistral**: Seems they're available internally, but aren't returned via the API

Let's leave it as an OpenAI specific setting for now, can be promoted another time (e.g. if AzureOpenAI is added as a discrete provider)